web/files/files.go

```diff
@@ -208,46 +218,61 @@ func ModificationHandler(c *gin.Context) {
 	}
 
 	patchData := container.Data
-	docType, err := vfs.ParseDocType(patchData.Type)
+
+	fileID := c.Param("file-id")
+
+	var docType vfs.DocType
+	if fileID == MetadataPath {
```

Comment:
Wow, this appears to be an heated discussion on both gin and the underlying httprouter. It should be solved by httprouter v2 (no timeline) or when/if we move to echo. 

Meanwhile however, it might be a good idea to be futurerproof and have the code be closer to the intent, may be with a anonymous function to do the split. something like
```go
router.GET("/:file-id", func (c *gin.Context) {
// dropping the constant as we only need it once and it's part of routing
 if c.Param("file-id") == "metadata" {      ModificationMetaHandler(c) 
 }else{ ModificationContentHandler(c) }
});
```